<!-- Launch file for the rrt-detector and the assigner -->


<launch>
<arg name="eta" value="1.0"/>
<arg name="Geta" value="15.0"/>
<arg name="namespace" value="tb3_1"/>
<arg name="force_gain_coef" value="1.5"/>

<param name="safety_distance" value="500"/>
<param name="percentage_threshold" value="24"/>
<param name="namespace_init_count" value="1"/>



  <node pkg="rrt_exploration" type="global_rrt_detector" name="global_detector" output="screen">
  <param name="eta" value="$(arg Geta)"/>
  <param name="map_topic" value="$(arg namespace)/map"/>
  <param name="force_gain_coeff" value="$(arg force_gain_coef)"/>
  </node>
  
  <node pkg="rrt_exploration" type="local_rrt_detector" name="local_detector" output="screen">
  <param name="eta" value="$(arg eta)"/>
  <param name="map_topic" value="$(arg namespace)/map"/>
  <param name="robot_frame" value="$(arg namespace)/base_link"/>
    <param name="force_gain_coeff" value="$(arg force_gain_coef)"/>
  </node>

  
  <node pkg="rrt_exploration" type="filter.py" name="filter" output="screen">
  <param name="map_topic" value="$(arg namespace)/map"/>
  <param name="info_radius" value="1"/> 
  <param name="costmap_clearing_threshold" value="70"/> 
  <param name="goals_topic" value="/detected_points"/>
  <param name="namespace" value="tb3_"/> 
  <param name="n_robots" value="1"/>
  <param name="rate" value="100"/>
  </node>
  

    <node pkg="rrt_exploration" type="explored_centroid.py" name="explored_centroid" output="screen">
        <param name="map_topic" value="$(arg namespace)/map"/> 
        <param name="rate" value="100"/> <!-- Set the update rate for the centroid calculation -->
    </node>




  <node pkg="rrt_exploration" type="assigner.py" name="assigner" output="screen">
  <param name="map_topic" value="$(arg namespace)/map"/>
  <param name="global_frame" value="map"/>
  <param name="info_radius" value="1"/> 
  <param name="info_multiplier" value="3.0"/> 
  <param name="hysteresis_radius" value="3.0"/> 
  <param name="hysteresis_gain" value="2.0"/> 
  <param name="frontiers_topic" value="/filtered_points"/> 
  <param name="n_robots" value="1"/>
  <param name="namespace" value="tb3_"/>
  <param name="delay_after_assignement" value="0.5"/>
  <param name="rate" value="100"/>
  </node>
  
  
  <node pkg="rrt_exploration" type="map_saver_node.py" name="map_saver_node" output="screen" required="true">
    <param name="map_topic" value="$(arg namespace)/map"/> <!-- 设置地图话题 -->
  </node>

          <!-- 运行velocity_adjust.py脚本 --> 
          <node pkg="rrt_exploration" type="velocity_adjust.py" name="velocity_adjust" output="screen" required="true">
            <param name="max_linear_speed" value="5.0"/> <!-- 设置最大线速度-->
            <param name="max_angular_speed" value="3"/> <!-- 设置最大角速度-->
            <param name="exploration_percentage_scale" value="0.0075"/> <!-- 设置探索百分比的缩放因子 -->
          </node>




</launch>
